components/match2/wikis/valorant/match_group_input_custom.lua

```diff
@@ -76,28 +78,175 @@ end
+	if data.finished == 'skip' or data.finished == 'np' or data.finished == 'cancelled' or data.finished == 'canceled' then
+		data.resulttype = 'np'
+	-- Map or Match is marked as finished.
+	-- Calculate and set winner, resulttype, placements and walkover (if applicable for the outcome)
+	elseif Logic.readBool(data.finished) then
+		if CustomMatchGroupInput.placementCheckDraw(indexedScores) then
+			data.winner = 0
+			data.resulttype = 'draw'
+			indexedScores = CustomMatchGroupInput.setPlacement(indexedScores, data.winner, 1, 1)
+		elseif CustomMatchGroupInput.placementCheckSpecialStatus(indexedScores) then
+			data.winner = CustomMatchGroupInput.getDefaultWinner(indexedScores)
+			data.resulttype = 'default'
+			if CustomMatchGroupInput.placementCheckFF(indexedScores) then
+				data.walkover = 'ff'
+			elseif CustomMatchGroupInput.placementCheckDQ(indexedScores) then
+				data.walkover = 'dq'
+			elseif CustomMatchGroupInput.placementCheckWL(indexedScores) then
+				data.walkover = 'l'
+			end
+			indexedScores = CustomMatchGroupInput.setPlacement(indexedScores, data.winner, 1, 2)
+		else
+			--Valorant only has exactly 2 opponents, neither more or less
+			if #indexedScores ~= 2 then
+				error('Unexpected number of opponents when calculating map winner')
+			end
+			if tonumber(indexedScores[1].score) > tonumber(indexedScores[2].score) then
+				data.winner = 1
+			else
+				data.winner = 2
+			end
+			indexedScores = CustomMatchGroupInput.setPlacement(indexedScores, data.winner, 1, 2)
+		end
+	end
+	return data, indexedScores
+end
+
+
+-- Check if any team has a none-standard status
+function CustomMatchGroupInput.placementCheckSpecialStatus(table)
+	return Table.any(table, function (_, scoreinfo) return scoreinfo.status ~= 'S' end)
+end
+
+-- function to check for forfeits
+function CustomMatchGroupInput.placementCheckFF(table)
+	return Table.any(table, function (_, scoreinfo) return scoreinfo.status == 'FF' end)
+end
+
+-- function to check for DQ's
+function CustomMatchGroupInput.placementCheckDQ(table)
+	return Table.any(table, function (_, scoreinfo) return scoreinfo.status == 'DQ' end)
+end
+
+-- function to check for W/L
+function CustomMatchGroupInput.placementCheckWL(table)
+	return Table.any(table, function (_, scoreinfo) return scoreinfo.status == 'L' end)
+end
+
+-- Get the winner when resulttype=default
+function CustomMatchGroupInput.getDefaultWinner(table)
+	for index, scoreInfo in pairs(table) do
+		if scoreInfo.status == 'W' then
+			return index
+		end
 	end
+	return -1
 end
 
 --
 -- match related functions
 --
+-- Calculate the match scores based on the map results.
+-- If it's a Best of 1, we'll take the exact score of that map
+-- If it's not a Best of 1, we should count the map wins
+-- Only update a teams result if it's
+-- 1) Not manually added
+-- 2) At least one map has a winner
+function matchFunctions.getScoreFromMapWinners(match)
+	-- For best of 1, display the results of the single map
```

Comment:
This line comment doesn't make sense for the next line. Where does it apply?

Reply:
Moved it to its correct place, 5 lines down.